guacamol/benchmark_suites.py

```diff
@@ -10,16 +11,313 @@
     kldiv_benchmark, perindopril_rings, amlodipine_rings, sitagliptin_replacement, zaleplon_with_other_formula, valsartan_smarts, \
     median_tadalafil_sildenafil, decoration_hop, scaffold_hop, ranolazine_mpo, pioglitazone_mpo
 
-
 def goal_directed_benchmark_suite(version_name: str) -> List[GoalDirectedBenchmark]:
-    if version_name == 'v1':
-        return goal_directed_suite_v1()
-    if version_name == 'v2':
-        return goal_directed_suite_v2()
-    if version_name == 'trivial':
-        return goal_directed_suite_trivial()
+    return GoalDirectedBenchmarkSuite().getBenchmark(version_name).getList()
+
+# factory (as singleton software engineering pattern)
+class GoalDirectedBenchmarkSuite(object):
```

Comment:
I fully support the addition of such classes, but I don't see the need for a complex factory / singleton registration in such a simple case. 

Why not something like:
```python
registered_goal_directed_benchmarks = {}

def register_goal_directed_benchmark(name, benchmark_suite_cls):
    # [...] some checks, preprocessing, etc.
    registered_goal_directed_benchmarks[name] = benchmark_suite_cls

class MyGoalDirectedBenchmarkSuite(GoalDirectedBenchmarkSuite):
    [...]

register_goal_directed_benchmark('my_suite_v1', MyGoalDirectedBenchmarkSuite)
```

I am happy to update this myself in the PR.